--- build/configure.ac.orig	2015-11-19 21:26:45.000000000 -0500
+++ build/configure.ac		2016-01-21 10:41:50.140727739 -0500
@@ -419,17 +419,26 @@
 AC_DEFUN([AC_C_ENDIAN],
 [AC_CACHE_CHECK(for endianness, ac_cv_c_endian,
 [
-  AC_RUN_IFELSE(
-    [AC_LANG_PROGRAM([], [dnl
+  if test $ac_compiler_gnu = yes; then
+    AC_MSG_CHECKING([with GNU flags])
+    if $CC -E -dM - < /dev/null | grep LITTLE_ENDIAN > /dev/null; then
+      ac_cv_c_endian=little
+    else
+      ac_cv_c_endian=big
+    fi
+  else
+    AC_RUN_IFELSE(
+      [AC_LANG_PROGRAM([], [dnl
         long val = 1;
         char *c = (char *) &val;
         exit(*c == 1);
+      ])
+    ],[
+      ac_cv_c_endian=big
+    ],[
+      ac_cv_c_endian=little
     ])
-  ],[
-    ac_cv_c_endian=big
-  ],[
-    ac_cv_c_endian=little
-  ])
+  fi
 ])
 if test $ac_cv_c_endian = big; then
   AC_DEFINE(ENDIAN_BIG, 1, [machine is bigendian])
